api/net/Areas/Services/Controllers/UserController.cs

```diff
@@ -48,14 +49,11 @@ public UserController(IUserService service)
     [HttpGet("{id}")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(UserModel), (int)HttpStatusCode.OK)]
-    [ProducesResponseType((int)HttpStatusCode.NoContent)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "User" })]
     public IActionResult FindById(int id)
     {
-        var result = _service.FindById(id);
-        if (result == null)
-            return NoContent();
-
+        var result = _service.FindById(id) ?? throw new NoContentException();
```

Comment:
why throw an exception here? Seems like a valid scenario that we might not have a user with a specific id.  Feels like we are pushing the responsibility back to the caller to ensure they only send valid ids?

Reply:
I generally prefer to return a 204, however it does result in more work on the client.  Which means by throwing a 400 we are actually giving the client less work since it already handles errors.  It also doesn't matter whether the caller gets a 204 or a 400, it's still their responsibility to request valid content.